diff --git a/build/gen.py b/build/gen.py
index 232e5363..b5020b15 100755
--- a/build/gen.py
+++ b/build/gen.py
@@ -469,6 +469,9 @@ def WriteGNNinja(path, platform, host, options, args_list):
         '-std=c++17'
     ])
 
+    if platform.is_zos():
+      cflags.extend(['-Wno-unknown-pragmas'])
+
     # flags not supported by gcc/g++.
     if cxx == 'clang++':
       cflags.extend(['-Wrange-loop-analysis', '-Wextra-semi-stmt'])
@@ -511,6 +514,7 @@ def WriteGNNinja(path, platform, host, options, args_list):
       cflags.append('-Wno-unused-function')
       cflags.append('-D_OPEN_SYS_FILE_EXT')
       cflags.append('-DPATH_MAX=1024')
+      cflags.append('-DZOSLIB_OVERRIDE_CLIB')
 
     if platform.is_posix() and not platform.is_haiku():
       ldflags.append('-pthread')
