diff --git a/build/gen.py b/build/gen.py
index 9168035b..7e420980 100755
--- a/build/gen.py
+++ b/build/gen.py
@@ -206,8 +206,8 @@ def main(argv):
                       default='../third_party/zoslib',
                       dest='zoslib_dir',
                       help=('Specify the path of ZOSLIB directory, to link ' +
-                            'with <ZOSLIB_DIR>/install/lib/libzoslib.a, and ' +
-                            'add -I<ZOSLIB_DIR>/install/include to the compile ' +
+                            'with <ZOSLIB_DIR>/lib/libzoslib.a, and ' +
+                            'add -I<ZOSLIB_DIR>/include to the compile ' +
                             'commands. See README.md for details.'))
 
   args_list.add_to_parser(parser)
@@ -412,7 +412,7 @@ def WriteGNNinja(path, platform, host, options, args_list):
       '.',
   ]
   if platform.is_zos():
-    include_dirs += [ options.zoslib_dir + '/install/include' ]
+    include_dirs += [ options.zoslib_dir + '/include' ]
 
   libs = []
 
@@ -491,6 +491,9 @@ def WriteGNNinja(path, platform, host, options, args_list):
         '-std=c++20'
     ])
 
+    if platform.is_zos():
+      cflags.extend(['-Wno-unknown-pragmas'])
+
     if is_gcc(cxx):
       cflags.append('-Wno-redundant-move')
       # https://gcc.gnu.org/bugzilla/show_bug.cgi?id=104336
@@ -905,7 +908,7 @@ def WriteGNNinja(path, platform, host, options, args_list):
     ])
 
   if platform.is_zos():
-    libs.extend([ options.zoslib_dir + '/install/lib/libzoslib.a' ])
+    libs.extend([ options.zoslib_dir + '/lib/libzoslib.a' ])
 
   if platform.is_windows():
     static_libraries['base']['sources'].extend([
